### PR TITLE
Install Python and Java from source

### PR DESCRIPTION
**Pull request by chansuke**
Fixes https://github.com/kubernetes-sigs/jobset/issues/190

Update the scripts to install Java and Python from source.

### CODE REVIEW

File: hack/python-sdk/gen-sdk.sh [hunk 1]
```diff
@@ -44,8 +44,27 @@ fi
 
 if [ -z `which java` ]; then
   echo "Installing OpenJDK 11"
+
   apt update -y
-  apt-get install openjdk-11-jdk -y
+
+  apt-get install -y wget unzip zip git autoconf build-essential \
```
Comment (danielvegamyhre) [inline hack/python-sdk/gen-sdk.sh:50]: The idea of installing from source was to decouple the installation process from environment specific package managers like `apt` - any way we can do this without using apt?

**Review by danielvegamyhre — CHANGES_REQUESTED** [review]
The idea of installing from source was to decouple the installation process from environment specific package managers like apt - any way we can do this without using apt?

File: hack/python-sdk/gen-sdk.sh (hunk 1 above)
Comment (chansuke) [reply hack/python-sdk/gen-sdk.sh:50]: @danielvegamyhre 
Thank you for your review. I will remove package-manager related part.

**Comment by danielvegamyhre** [comment]
Thanks for working on this, it may address #193 as well

**Comment by chansuke** [comment]
@danielvegamyhre 
Thank you so much for your comment.

All the tests are passing except for `pull-jobset-test-unit-main`. The error message that I am seeing is:
> make: *** [Makefile:118: test-python-sdk] Error 1

I am not able to replicate this error on my local machine. Do you have any suggestions on how I could debug this issue further?

**Comment by kannon92** [comment]
Maybe we should look into using a container for generation?

Only way I can think of to avoid doing apt on a user machine and having a more stable build.

**Comment by danielvegamyhre** [comment]
@chansuke Try adding `set -x` to the top of `test-sdk.sh` file near the other `set -o ...` lines. This should enable you to see what line is causing the error in the logs.

**Comment by danielvegamyhre** [comment]
@chansuke are you still working on this? I think @kannon92 is right, we should look into using a container for generation if possible, as we may be able to get downloading and building from source to work in CI and some of our dev environments, but inevitably there will be contributors whose environment will be incompatible with the specific Java/Python builds we're doing here.

**Comment by danielvegamyhre** [comment]
Closing this due to inactivity

**Pull request closed without merging**